creator-node/src/premiumContent/PremiumContentQueue.ts

```diff
@@ -0,0 +1,127 @@
+ */
+export class PremiumContentQueue {
+  queue: Queue
+
+  constructor() {
+    const connection = {
+      host: config.get('redisHost'),
+      port: config.get('redisPort')
+    }
+    this.queue = new Queue(QUEUE_NAME, {
+      connection,
+      defaultJobOptions: {
+        removeOnComplete: PREMIUM_CONTENT_QUEUE_HISTORY,
+        removeOnFail: PREMIUM_CONTENT_QUEUE_HISTORY
+      }
+    })
+    this.queue.drain(true)
+    ;(() =>
+      new Worker(
+        QUEUE_NAME,
+        async (_) => {
+          await this.logStatus('Starting')
+          try {
+            const premiumContentCIDMap = await this.getPremiumContentCIDMap()
+            await updatePremiumContentCIDCache({
+              cacheMap: premiumContentCIDMap,
+              logger
+            })
+          } catch (e) {
+            this.logStatus(
+              `Error in premium content worker: ${(e as Error).message}`
+            )
+          }
+        },
+        { connection }
+      ))()
+  }
+
+  /**
+   * Joins the "Tracks" and "Files" tables on track ids and sees if the
+   * corresponding track is premium.
+   * Returns map of these premium CIDs to their corresponding track ids.
+   */
+  async getPremiumContentCIDMap() {
+    const result = await models.sequelize.query(
+      `select t."blockchainId", t."metadataJSON", f."multihash" from "Tracks" t join "Files" f
+        on t."blockchainId" = f."trackBlockchainId"
+        where f."type" in ('track', 'copy320')`,
+      {
+        type: QueryTypes.SELECT
+      }
+    )
+    const nonPremiumCIDSet = new Set()
+    const cidMap: { [key: string]: number } = {}
+    result.forEach(
+      (record: {
+        blockchainId: number
+        metadataJSON: any
+        multihash: string
+      }) => {
+        if (record.metadataJSON.is_premium) {
+          if (!nonPremiumCIDSet.has(record.multihash)) {
+            // Only add to CID to map if we have not yet seen
+            // a non-premium track with the same CID.
+            cidMap[record.multihash] = record.blockchainId
+          }
+        } else if (cidMap[record.multihash]) {
+          // If the same CID exists in the map, remove it from the map
+          // because there exists a non-premium track with the same CID.
+          // Also add it to the non-premium CID set to make sure
+          // this CID never gets re-entered in the map.
+          delete cidMap[record.multihash]
+          nonPremiumCIDSet.add(record.multihash)
+        }
+      }
+    )
+    return cidMap
+  }
+
+  /**
+   * Logs a status message and includes current queue info
+   */
+  async logStatus(message: string) {
+    const { waiting, active, completed, failed, delayed } =
+      await this.queue.getJobCounts()
+    logger.info(
+      `Premium Content Queue: ${message} || active: ${active}, waiting: ${waiting}, failed ${failed}, delayed: ${delayed}, completed: ${completed} `
+    )
+  }
+
+  /**
+   * Starts the premium content queue
+   */
+  async start() {
```

Comment:
we should surround all of this in `if (clusterUtils.isThisWorkerSpecial()) {` so that not all cluster workers add jobs